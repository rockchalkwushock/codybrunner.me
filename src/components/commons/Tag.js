import { InternalLink } from './Link'

const Tag = InternalLink.extend`
  align-items: center;
  background-color: ${({ theme }) => theme.site.fg};
  border: none;
  border-radius: 10px;
  box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.7);
  cursor: pointer;
  display: flex;
  font-size: 0.8rem;
  justify-content: center;
  letter-spacing: 1px;
  margin: 0.25em 0.5em;
  outline: none;
  padding: 0.25em 0.5em;
  transform: scale(1);
  transition: all 0.3s ease-in-out;
  :hover {
    color: ${({ theme }) => theme.site.linkClr};
  }
`

Tag.displayName = 'Tag'

export default Tag
