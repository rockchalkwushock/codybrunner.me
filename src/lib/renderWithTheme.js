import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
