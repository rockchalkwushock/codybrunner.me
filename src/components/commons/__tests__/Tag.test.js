import React from 'react'

import renderWithTheme from '../../../lib/renderWithTheme'
import Tag from '../Tag'

// FIXME: Issue with `history undefined` coming from HOC.
// I beleive props are not being passed through properly.
describe.skip('Component: <Tag />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<Tag to="/tags/javascript">javascript</Tag>)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('align-items', 'center')
    expect(wrapper).toHaveStyleRule('background-color', '')
    expect(wrapper).toHaveStyleRule('border', 'none')
    expect(wrapper).toHaveStyleRule('border-radius', '10px')
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '1px 1px 5px 1px rgba(0, 0, 0, 0.7)'
    )
    expect(wrapper).toHaveStyleRule('cursor', 'pointer')
    expect(wrapper).toHaveStyleRule('display', 'flex')
    expect(wrapper).toHaveStyleRule('font-size', '0.7rem')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('letter-spacing', '1px')
    expect(wrapper).toHaveStyleRule('outline', 'none')
    expect(wrapper).toHaveStyleRule('padding', '0.25em 0.5em')
    expect(wrapper).toHaveStyleRule('transition', 'all 0.3s ease-in-out')
    expect(wrapper).toHaveStyleRule('color', '#ffe1b6', {
      modifier: ':hover'
    })
  })
})
