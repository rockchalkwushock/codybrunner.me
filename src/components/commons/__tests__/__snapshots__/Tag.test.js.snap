// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Tag /> should render correctly 1`] = `undefined`;
